### ISSUE TITLE
Pincode bypassable under certain conditions

### ISSUE THREAD

**Issue by Eyco1981** [nielsfaber/alarmo#329]
### Checklist
- [X] I checked for similar existing issues (both open and closed) before posting.
- [X] I am running the latest version.

### Expected behavior
Pincode is always needed (if settings are accordingly) to disarm Alarmo.

### Actual behavior
When Alarmo is in armed_home and a service call (e.g. by using a keypad) switches it to armed_away mode, it returns to mode disarmed without asking for a pincode when the arming fails (e.g. an open door sensor, which is fine for armed_home, but has to be closed to be able to arm armed_away).
This way I can disarm Alarmo in armed_home by opening that door and trigger a mode change to armed_away on the keypad. The change fails due to the open door and Alarmo is disarmed without pincode.

### Steps to Reproduce
- Configure Alarmo to require a pincode to disarm, but none to arm
- Arm Alarmo in armed_home mode
- Trigger a sensor (e.g. door sensor) which may be open in armed_home mode but has to be closed to arm mode armed_away
- Switch from armed_home to armed_away by service call

**Comment by nielsfaber** [comment]
If the transition from armed_home to armed_away fails, the alarm should revert to armed_home and certainly not disarmed.
Thanks for reporting this issue.
I will pick it up with priority since this sounds like it could compromise security.

**Comment by nielsfaber** [comment]
I could reproduce this problem. It only occurs for the master alarm, when arming separate areas it does not happen.
I also found the code responsible for the behaviour and I will work on a solution.
Thanks for reporting this 👍

**Comment by nielsfaber** [comment]
I made several tests and improvements to handle the different cases.
The problems are now solved. I will let you know when a next release is out.
Note that as part of the changes, the exit delay will be used when switching from armed_home -> armed_away (and others), and also user needs to enter the code for doing so.
These changes might cause some trouble for existing users but I do think they are needed for being consistent.
